server/plugin.go

```diff
@@ -221,81 +226,155 @@ func (p *Plugin) startSubscriptions() {
 		return
 	}
 
+	mmChannelSubscriptions, err := p.store.ListChannelSubscriptions()
+	if err != nil {
+		p.API.LogError("Unable to list Mattermost channel subscriptions", "error", err)
+		return
+	}
+
+	mmGlobalSubscriptions, err := p.store.ListGlobalSubscriptions()
+	if err != nil {
+		p.API.LogError("Unable to list Mattermost global subscriptions", "error", err)
+		return
+	}
+
+	msteamsSubscriptionsMap, allChatsSubscription, err := p.monitor.GetMSTeamsSubscriptionsMap()
+	if err != nil {
+		p.API.LogError("Unable to list MS Teams subscriptions", "error", err)
+		return
+	}
+
+	mmChannelSubscriptionsMap := make(map[string]storemodels.ChannelSubscription)
+	for _, mmChannelSubscription := range mmChannelSubscriptions {
+		mmChannelSubscriptionsMap[mmChannelSubscription.SubscriptionID] = mmChannelSubscription
+	}
+
 	wg := sync.WaitGroup{}
 	ws := make(chan struct{}, 20)
 
 	wg.Add(1)
 	ws <- struct{}{}
 	go func() {
 		defer wg.Done()
-		chatsSubscription, err := p.msteamsAppClient.SubscribeToChats(p.GetURL()+"/", p.getConfiguration().WebhookSecret, !p.getConfiguration().EvaluationAPI)
-		if err != nil {
-			p.API.LogError("Unable to subscribe to chats", "error", err)
-			// Mark this subscription to be created and retried by the monitor system
-			_ = p.store.SaveGlobalSubscription(storemodels.GlobalSubscription{
-				SubscriptionID: "fake-subscription-id",
-				Type:           "allChats",
-				ExpiresOn:      time.Now(),
-				Secret:         p.getConfiguration().WebhookSecret,
-			})
-			<-ws
-			return
-		}
-		p.API.LogDebug("Subscription to all chats created", "subscriptionID", chatsSubscription.ID)
+		// Check if all chats subscription is present on Mattermost
```

Comment:
```suggestion
		// Check if all chat subscriptions are present on Mattermost
```
I think this is better. If changing then change the variable names also

Reply:
we are creating a single subscription for all the chats whose type is `allChats`. So, I think the current comment is fine already.